sdk/servicebus/azure-servicebus/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,1335 @@
+        # Process message
+        receiver.complete_message(message)
+```
+
+3. Handle lock lost errors gracefully by catching the exception and potentially re-receiving the message
+
+### Message size issues
+
+**MessageSizeExceededError resolution:**
+1. Reduce message payload size
+2. Use message properties and application properties for metadata instead of body
+3. For batch operations, reduce the number of messages in the batch
+4. Consider splitting large messages across multiple smaller messages
+
+**Service Bus message size limits:**
+- Standard tier: 256 KB per message
+- Premium tier: 1 MB per message
+
+For the most up-to-date information on Service Bus limits, refer to the [Azure Service Bus quotas and limits](https://learn.microsoft.com/azure/service-bus-messaging/service-bus-quotas) documentation.
+
+### Message settlement issues
+
+**MessageAlreadySettled resolution:**
+1. Ensure you're not trying to settle the same message multiple times
+2. Check your application logic for race conditions
+3. Use try-catch blocks when settling messages
+
+```python
+try:
+    receiver.complete_message(message)
+except MessageAlreadySettled:
+    # Message was already settled, this is expected in some scenarios
+    pass
+```
+
+### Dead letter queue issues
+
+Messages can be moved to the dead letter queue for various reasons:
+
+**Common reasons:**
+- Message TTL expired
+- Max delivery count exceeded
+- Message was explicitly dead lettered
+- Message processing failed repeatedly
+
+**Debugging dead letter messages:**
+```python
+# Receive from dead letter queue
+dlq_receiver = servicebus_client.get_queue_receiver(
+    queue_name="your_queue", 
+    sub_queue=ServiceBusSubQueue.DEAD_LETTER
+)
+
+with dlq_receiver:
+    messages = dlq_receiver.receive_messages(max_message_count=10)
+    for message in messages:
+        print(f"Dead letter reason: {message.dead_letter_reason}")
+        print(f"Dead letter description: {message.dead_letter_error_description}")
+```
+
+## Troubleshooting session handling issues
+
+### Session lock issues
+
+Session-enabled entities require proper session management.
+
+**SessionLockLostError resolution:**
+1. Renew session locks before they expire
+2. Use `AutoLockRenewer` for automatic session lock renewal
+3. Handle session lock lost errors by reconnecting to the session
+
+```python
+from azure.servicebus import AutoLockRenewer
+
+renewer = AutoLockRenewer()
+with receiver:
+    session = receiver.session
+    renewer.register(receiver, session, max_lock_renewal_duration=300)
+    # Process messages in session
+```
+
+### Session cannot be locked
+
+**SessionCannotBeLockedError resolution:**
+1. Ensure no other clients are already connected to the same session
+2. Wait for the current session lock to expire before reconnecting
+3. Use a different session ID if specific session is not required
+
+## Troubleshooting sender issues
+
+### Cannot send batch with multiple partition keys
```

Comment:
@kashifkhan - I don't think this is common. We can probably delete?

Reply:
yes we can delete. Its in the samples, doesnt need to be here